steps.go

```diff
@@ -46,7 +46,7 @@ func (r Steps) Last() uint64 {
 func (r Steps) AsSlice() []uint64 {
 	if r.isSequential {
 		last := r.Last()
-		steps := []uint64{}
+		steps := make([]uint64, 0, last)
```

Comment:
By passing a third argument into make, you can define a slice of length `0 (second argument)` with room for `last (third argument)` more.
https://play.golang.org/p/mB8prfjjrIK

(It might not be needed to fix this if a performance bottleneck isn't here.)